app/uk/gov/hmrc/brm/connectors/BirthConnector.scala

```diff
@@ -40,35 +40,33 @@ trait BirthConnector {
 
   /** RequestType, reference or details
     */
-  trait RequestType
-
-  case class ReferenceRequest() extends RequestType
-
-  case class DetailsRequest() extends RequestType
+  sealed trait RequestType
+  case object ReferenceRequest extends RequestType
+  case object DetailsRequest extends RequestType
 
   case class Request(uri: String, jsonBody: JsValue)
 
   private def buildRequest(payload: Payload, operation: RequestType): Request = {
     val f       = operation match {
-      case ReferenceRequest() => referenceBody
-      case DetailsRequest()   => detailsBody
+      case ReferenceRequest => referenceBody
+      case DetailsRequest   => detailsBody
     }
     val request = f(payload)
     Request(request._1, request._2)
   }
 
   private def sendRequest(request: Request)(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[HttpResponse] = {
 
-    val newHc = hc.copy(authorization = None)
+    implicit val headerCarrier: HeaderCarrier = hc.copy(authorization = None)
```

Comment:
Personal preference but I dont think you need this to be implicit if you are parsing it explicitly as one of the args to `.post()(headerCarrier)`, could also just move this down to the post method e.g.
```scala annotate
    val response = http
      .post(url"${request.uri}")(hc.copy(authorization = None))
      ...
```

Reply:
Amended